lms/services/exceptions.py

```diff
@@ -12,26 +12,31 @@ class ExternalRequestError(Exception):
     :type details: JSON-serializable dict or ``None``
     """
 
-    def __init__(self, message="External request failed", response=None, details=None):
+    def __init__(self, message=None, response=None, details=None):
         super().__init__()
         self.message = message
         self.response = response
         self.details = details
 
+    def __repr__(self):
+        if self.response:
+            # Include the details of the response for debugging.
+            # This appears in the logs and in tools like New Relic and Sentry.
+            response = (
+                "Response("
+                f"status_code={repr(self.response.status_code)}, "
+                f"reason={repr(self.response.reason)}, "
+                f"text={repr(self.response.text)})"
+            )
+        else:
+            response = "None"
+
+        return (
+            f"ExternalRequestError(message={repr(self.message)}, response={response})"
+        )
```

Comment:
Changed this from `__str__()` to `__repr__()` because `__repr__()` is used by the debugger (so now we get to see the nice string representation in the debugger as well).